In [6]:
import pandas as pd
import numpy as np


In [7]:
data = pd.read_csv(r"C:\Users\gbeme\Desktop\Works\Food_Bal_Sheet\FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding='latin-1')
data.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [8]:
data.columns
years=['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018']

In [20]:
#11. What is the total sum of Animal Fat produced in 2014 and 2017 respectively?
anifat_sum1417 = data.groupby('Item')[['Y2014', 'Y2017']].sum().loc['Animal fats']
anifat_sum1417

Y2014    209460.54
Y2017    269617.53
Name: Animal fats, dtype: float64

In [21]:
#12. What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?
mean_std15 = round(data.describe()[['Y2015']], 3).loc[['mean','std']]
mean_std15
#or
#mean15 = round(data['Y2015'].mean(),3)
#std15 = round(data['Y2015'].std(),3)
#print(mean15, std15, sep=', ')

Y2015
mean   135.236
std   1603.404

In [11]:
#13. What is the total number and percentage of missing data in 2016 to 2 decimal places?
total_missing16=data['Y2016'].isna().sum()
percent_miss16=round((data['Y2016'].isna().sum()/len(data['Y2016']))*100, 2)
print(total_missing16)
print(percent_miss16)
print(f"The total number of missing data in 2016 is {total_missing16}")
print(f"{percent_miss16}% of the data in 2016 was missing")

1535
2.52
The total number of missing data in 2016 is 1535
2.52% of the data in 2016 was missing


In [27]:
# 14. Which year had the highest correlation with ‘Element Code’? 
eleyear_corr = data[years].corrwith(data['Element Code']).sort_values(ascending=False)
eleyear_corr

Y2014    0.024457
Y2018    0.024279
Y2017    0.024254
Y2015    0.023889
Y2016    0.023444
dtype: float64

Year 2014 has the highest correlation with element code (0.02445)

In [13]:
# 15. What year has the highest sum of Import Quantity?
total_import=data.groupby('Element').sum().loc['Import Quantity']
total_import[years].sort_values(ascending=False)

Y2017    294559.09
Y2018    287997.09
Y2016    286582.78
Y2014    274144.48
Y2015    267018.46
Name: Import Quantity, dtype: float64

2017 is the year with the highest sum of Import Quantity

In [14]:
#16. What is the total number of the sum of Production in 2014?
prod_sum14 = data.groupby('Element').sum()[['Y2014']].loc['Production']
prod_sum14

Y2014    1931287.75
Name: Production, dtype: float64

The total production for 2014 is 1.93 million

In [15]:
#17. Which of these elements had the highest sum in 2018?
tot_element18 = data.groupby('Element').sum()['Y2018']

tot_element18.sort_values(ascending=False)[:3]

Element
Domestic supply quantity    2161192.10
Production                  2075072.89
Food                        1303841.28
Name: Y2018, dtype: float64

The Domestic supply quantity has the highest sum in 2018



In [16]:
#18. Which of these elements had the 3rd lowest sum in 2018?
tot_element18.sort_values()[:3]

Element
Tourist consumption                          90.00
Fat supply quantity (g/capita/day)        10258.69
Protein supply quantity (g/capita/day)    11833.56
Name: Y2018, dtype: float64

The protein supply has the 3rd lowest sum in 2018

In [17]:
#19. What is the total Import Quantity in Algeria in 2018?
alg_data = data[data.Area=='Algeria']
alg_data.groupby('Element').sum()['Y2018'].loc[['Import Quantity']]

Element
Import Quantity    36238.29
Name: Y2018, dtype: float64

The total import quantity for Algeria in 2018 is 36238

In [28]:
#20.What is the total number of unique countries in the dataset?
area_nunique = data['Area'].nunique()
area_nunique

49

There are 49 unique countries in the data set